File: appambit-sdk/src/main/java/com/appambit/sdk/core/utils/DateUtils.java
package com.appambit.sdk.core.utils;

import androidx.annotation.NonNull;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.TimeZone;

public class DateUtils {
    public static Date getUtcNow() {
        Calendar calendar = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
        return calendar.getTime();
    }

    @NonNull
    public static Date getDateDaysAgo(int daysAgo) {
        Calendar calendar = Calendar.getInstance(TimeZone.getTimeZone("UTC"));
        calendar.add(Calendar.DAY_OF_YEAR, -daysAgo);
        return calendar.getTime();
    }
}

File: tests/AppAmbitSdkTest/src/test/java/com/appambitsdk/test/unit/utils/DateUtils.kt
package com.appambitsdk.test.unit.utils

import java.util.Calendar
import java.util.Date
import java.util.TimeZone

class DateUtils {
    companion object{
        fun getUtcNow(): Date {
            val calendar = Calendar.getInstance(TimeZone.getTimeZone("UTC"))
            return calendar.getTime()
        }
    }
}